docs/GDPs/gdp-0002.rst

```diff
@@ -0,0 +1,346 @@
+
+  def ub(uc, vc, ub, dt4, dt5, partitioner):  # new SubtilePartitioner object
+    with partitioner.center(uc) as domain:
+        main_ub(uc,vc,grid.cosa,grid.rsina,ub,dt5=dt5,origin=domain.origin, domain=domain.extent,)
+    with partitioner.tile_west(ut, start=0, end=1) as domain:
+        x_edge(ut, ub, dt5=dt5, origin=domain.origin, domain=domain.extent)
+    with partitioner.tile_south(ut, start=0, end=1) as domain:
+        y_edge(ut, ub, dt4=dt4, origin=domain.origin, domain=domain.extent)
+    with partitioner.tile_north(ut, start=-1, end=0) as domain:
+        y_edge(ut, ub, dt4=dt4, origin=domain.origin, domain=domain.extent)
+    with partitioner.tile_east(ut, start=-1, end=0) as domain:
+        x_edge(ut, ub, dt5=dt5, origin=domain.origin, domain=domain.extent)
+
+This is similar to the idea of a region object, but is more directly utilizing a user defined object rather than a special boundary keyword.
+It is unclear how this would translate to the backend.
+
+
+Discussion
+----------
+
+This issue is related to this discussion, a precursor to this proposal:
+https://github.com/GridTools/gt4py/issues/9
+It is an alternative approach to the backend optimization of corners and edges.
+
+References and Footnotes
+------------------------
+
+.. [1] Each GDP must either be explicitly labeled as placed in the public domain (see
+   this GDP as an example) or licensed under the `Open Publication License`_.
+
+.. _Open Publication License: https://www.opencontent.org/openpub/
+
+
+Copyright
+---------
+
+This document has been placed in the public domain. [1]_
+
+Additional Examples
+-------------------
+
+Here are additional examples that may be helpful for discussion as we review this proposal.
+
+Special corner handling
++++++++++++++++++++++++
+
+.. code-block:: python
+
+  SWCorner(starti, endi, startj, endj) -> Box(slice(starti + global_istart, endi + global_istart), slice(startj + global_jstart, endj + global_jstart))​
+  SECorner(starti, endi, startj, endj) -> Box(slice(starti + global_iend, endi + global_iend), slice(startj + global_jstart, endj + global_jstart))​
+  NWCorner(starti, endi, startj, endj) -> Box(slice(starti + global_istart, endi + global_istart), slice(startj + global_jend, endj + global_jend))​
+  NECorner(starti, endi, startj, endj) -> Box(slice(starti + global_iend, endi + global_iend), slice(startj + global_jend, endj + global_jend))​
+
+  @gtscript.stencil(backend="dawn:gtmc", partitioner=partitioner)
+  def brbl_stencil(q: FIELD_TYPE, al: FIELD_TYPE, br: FIELD_TYPE, bl: FIELD_TYPE):
+    with computation(PARALLEL), interval(...):
+        br = al[0, 1, 0] - q
+    with boundary(SWCorner(-1, 0, 0, 1), SECorner(0, 1, 0, 1), NWCorner(-1, 0, 1, 2), NECorner(0, 1, 1, 2)), interval(...):
+        br = 0
+    with computation(PARALLEL), interval(...):
+        bl = br[-1, 0, 1]  + al
+
+  brbl_stencil(al, q, br, origin=(local_istart - 1, local_jstart, 0), domain=(local_ni + 1, local_nj, nk))
```

Comment:
I think corner handling is an obvious extension of the edge handling example written above, it shouldn't be necessary to include it.
```suggestion
```